release-tools/prow.sh

```diff
@@ -137,14 +137,23 @@ configvar CSI_PROW_BUILD_JOB true "building code in repo enabled"
 configvar CSI_PROW_KUBERNETES_VERSION 1.17.0 "Kubernetes"
 
 # This is a hack to workaround the issue that each version
-# of kind currently only supports specific patch versions of
-# Kubernetes. We need to override CSI_PROW_KUBERNETES_VERSION
+# of kind currently only has images for specific patch versions of
+# Kubernetes, as documented in the release notes of each
+# kind release.
+#
+# We need to override CSI_PROW_KUBERNETES_VERSION
```

Comment:
I already changed our CI jobs to explicitly specify the K8s version so we should update them there too: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-csi/gen-jobs.sh#L25

Reply:
I would like to work towards a goal where adding support for a new K8s version doesn't require updating the release framework to understand the new version. I'm looking at k/k and kubetest as an example. But maybe that is not achievable at the moment because of the kind image dependencies.

Reply:
Yes, kind is the problem here. There's just no clean "API" for installing kind and configuring it for different Kubernetes versions.

We either have to have all of that in the component or in the job definition. In the latter case, the prow.sh would have to be given some helper script that complete replaces the current `start_kubernetes`. The downside then will be that it becomes much harder to try out changes to prow.sh locally because one has to provide that helper script.